nf_core/pipeline-template/.github/actions/nf-test/action.yml

```
@@ -0,0 +1,114 @@
+name: "nf-test Action"
+description: "Runs nf-test with common setup steps"
+inputs:
+  profile:
+    description: "Profile to use"
+    required: true
+  shard:
+    description: "Shard number for this CI job"
+    required: true
+  total_shards:
+    description: "Total number of test shards(NOT the total number of matrix jobs)"
+    required: true
+  paths:
+    description: "Test paths"
+    required: true
+
+runs:
+  using: "composite"
+  steps:
+    - name: Setup Nextflow
+      uses: nf-core/setup-nextflow@v2
+      with:
+        version: "${{ env.NXF_VERSION }}"
+
+    - name: Set up Python
+      uses: actions/setup-python@42375524e23c412d93fb67b49958b491fce71c38 # v5
+      with:
+        python-version: "3.11"
+
+    - name: Install nf-test
+      uses: nf-core/setup-nf-test@v1
+      with:
+        version: "${{ env.NFT_VER }}"
+
+    - name: Setup apptainer
+      if: contains(inputs.profile, 'singularity')
+      uses: eWaterCycle/setup-apptainer@main
+
+    - name: Set up Singularity
+      if: contains(inputs.profile, 'singularity')
+      shell: bash
```

This can also be an exact match, right?
```suggestion
    - name: Setup apptainer
      if: ${{inputs.profile == 'singularity'}}
      uses: eWaterCycle/setup-apptainer@main

    - name: Set up Singularity
      if: ${{inputs.profile == 'singularity'}}
      shell: bash
```

---

I'd say no, if someone want to run singularity,test then it won't work with an exact match

---

ok, then we have to change it for conda too